File: fixtures/scale/haskell/large/Ciphers/Aes128/Impl.hs
module Ciphers.Aes128.Impl (encrypt, decrypt) where

import Ciphers.Aes128.Model (keyBits, rounds)

encrypt :: Int -> [Int] -> [Int]
encrypt key = map (\b -> mod (b + key + rounds + keyBits) 256)

decrypt :: Int -> [Int] -> [Int]
decrypt key = map (\b -> mod (b - key - rounds - keyBits + 512) 256)

File: fixtures/scale/elixir/large/lib/ciphers/xtea128/impl.ex
defmodule Ciphers.Xtea128.Impl do
  alias Ciphers.Xtea128.Model

  def encrypt(key, bytes) do
    Enum.map(bytes, &rem(&1 + key + Model.rounds() + Model.key_bits(), 256))
  end

  def decrypt(key, bytes) do
    Enum.map(bytes, &rem(&1 - key - Model.rounds() - Model.key_bits() + 512, 256))
  end
end
